File: apiteddymo/app/Models/Experience.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;

class Experience extends Model
{
    use HasUuids;
    
    protected $fillable = [
        'id', 'title', 'jobtitle', 'company', 'duration', 'description', 'link','is_deleted'
    ];

    protected $casts = [
        'technologies' => 'array', // Cast to array for JSON field
    ];

    public function responsibilities()
    {
        return $this->hasMany(Responsibility::class);
    }
}

File: DotnetAPI/Models/Experience.cs
namespace DotnetAPI.Models
{
    public class Experience
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public required string JobTitle { get; set; }
        public required string Company { get; set; }
        public required string Duration { get; set; }
        public required string Description { get; set; }
        public string? Link { get; set; }
        public bool IsActive { get; set; }
        public DateTime? DeletedDate { get; set; }

        public List<ExperienceResponsibility> Responsibilities { get; set; } = new List<ExperienceResponsibility>();
    }

    public class ExperienceResponsibility
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public required string Description { get; set; }
        public required string Technologies { get; set; }

        public Guid ExperienceId { get; set; }
        public required Experience Experience { get; set; }
    }
}
